src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -317,6 +322,64 @@ public Response replaceFileInDataset(
         }
             
     } // end: replaceFileInDataset
+
+    /**
+     * Delete an Existing File 
+     * 
+     * @param id file ID or peristent ID
+     */
+    @DELETE
+    @Path("{id}")
+    public Response deleteFileInDataset(@PathParam("id") String fileIdOrPersistentId){
+
+        if (!systemConfig.isHTTPUpload()) {
+            return error(Response.Status.SERVICE_UNAVAILABLE, BundleUtil.getStringFromBundle("file.api.httpDisabled"));
+        }
+        // (1) Get the user from the API key
+        User authUser;
+        try {
+            authUser = findUserOrDie();
```

Comment:
When you merge the latest from develop (please do!) you'll find that findUserOrDis can't be found. This is the error I got:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project dataverse: Compilation failure
[ERROR] /Users/pdurbin/github/iqss/dataverse/src/main/java/edu/harvard/iq/dataverse/api/Files.java:[335,24] cannot find symbol
[ERROR]   symbol:   method findUserOrDie()
[ERROR]   location: class edu.harvard.iq.dataverse.api.Files

I'm not 100% sure myself what the way is but I'm guessing it's something like this (based on other parts of Files.java after you merge):

```
DataverseRequest req;
try {
    req = createDataverseRequest(getRequestUser(crc));
} catch (Exception e) {
    return error(BAD_REQUEST, "Error attempting to request information. Maybe a bad API token?");
}
```

Please take a look and hack around. We can both learn a bit by studying PR where this change was introduced:

- #9303

Reply:
findUserOrDie was deprecated in the PR above and removed in this PR:

- #9360

Reply:
This is the javadoc that was deleted when findUserOrDie was deleted:

In case you are implementing a new endpoint that requires user authentication, here is an example of how to apply the new filter-based authentication:
{@link edu.harvard.iq.dataverse.api.Datasets#getDataset(ContainerRequestContext, String, UriInfo, HttpHeaders, HttpServletResponse)}

That code is here: https://github.com/IQSS/dataverse/blob/f97fc0567b0984f3be3069d7263c16168ee23061/src/main/java/edu/harvard/iq/dataverse/api/Datasets.java#L249

But I suppose other examples could be followed as well.

Reply:
methods that used the old API are now anotated @AuthRequired and have @Context  ContainerRequestContext crc
eg
    @POST
    @AuthRequired
    @Path("{id}/replace")
    @Consumes(MediaType.MULTIPART_FORM_DATA)
    public Response replaceFileInDataset(
                    @Context ContainerRequestContext crc,
                    @PathParam("id") String fileIdOrPersistentId,
                    @FormDataParam("jsonData") String jsonData,
                    @FormDataParam("file") InputStream testFileInputStream,
                    @FormDataParam("file") FormDataContentDisposition contentDispositionHeader,
                    @FormDataParam("file") final FormDataBodyPart formDataBodyPart
                    ){
...
       User authUser = getRequestUser(crc);

where AbstractApiMethod is
    protected User getRequestUser(ContainerRequestContext crc) {
        return (User) crc.getProperty(ApiConstants.CONTAINER_REQUEST_CONTEXT_USER);
    }

Reply:
I have merged the develop branch and added annotation/replaced getting user method.